code/game/objects/items/trash.dm

```diff
@@ -212,6 +212,18 @@
 	name = "candle"
 	icon = 'icons/obj/candle.dmi'
 	icon_state = "candle4"
+	starting_materials = list(MAT_WAX = (4*CC_PER_SHEET_WAX/5))
+	var/image/wick
+
+/obj/item/trash/candle/New(turf/loc, var/obj/item/candle/source)
+	..()
+	if (source)
```

Comment:
`color = source ? source.color : COLOR_DEFAULT_CANDLE`

Reply:
Unless we're working with a string, I'd argue that whether to use a ternary operator or not in a given case is best left to the coder's own discretion. I generally try to avoid using them so the code is easier to read for other people.